build/release-tars/BUILD.bazel

```diff
@@ -11,13 +11,19 @@ load(
 load("@io_k8s_repo_infra//defs:build.bzl", "release_filegroup")
 load("@io_k8s_repo_infra//defs:pkg.bzl", "pkg_tar")
 
-# Bazel doesn't make the output filename
-# (such as kubernetes-server-{OS}-{ARCH}.tar.gz) configurable, so we instead
-# create rules for all platforms and tag them manual.
-# We then select the correct set of platform-specific tarballs in this filegroup
-# using a select() statement.
-# Thus the release-tars target always selects the correct set of tarballs
-# for the configured platform being built.
+# Bazel doesn't make the output filename (such as
+# kubernetes-server-{OS}-{ARCH}.tar.gz) configurable, so we instead create rules
+# for all platforms and tag them manual. We then select the correct set of
+# platform-specific tarballs in this filegroup using a select() statement. Thus
+# the release-tars target always selects the correct set of tarballs for the
+# configured platform being built.
+
+# This rule is called by 'cmrel stage' to create all release artifacts for a
+# particular os/arch combination. OS and ARCH args get substituted in one of the
+# functions that 'for_platforms' call. That way a Bazel target name gets built
+# that corresponds to the output of one of the rules below and thus invokes
+# rules below to generate the actual artifacts (server images, manifests,
+# kubectl plugin binaries).
```

Comment:
praise: as with so many of the comments you've added, this is really useful :+1: